src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private int maxListSize = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[maxListSize];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growIfArrayFull();
+        size++;
```

Comment:
It's better to have two variables. Size - the number of occupied cells.
Capacity - the number of all cells.
In fact, it is better to remake the decision based on this information.